crates/rspack_plugin_css/src/plugin/impl_plugin_for_css_plugin.rs

```diff
@@ -36,39 +36,53 @@ impl CssPlugin {
     chunk: &Chunk,
     ordered_css_modules: &[&dyn Module],
   ) -> rspack_error::Result<ConcatSource> {
-    let module_sources = ordered_css_modules
-      .iter()
-      .map(|module| {
-        let module_id = &module.identifier();
-        let code_gen_result = compilation
-          .code_generation_results
-          .get(module_id, Some(&chunk.runtime));
-
-        Ok(
-          code_gen_result
-            .get(&SourceType::Css)
-            .map(|source| (CssModuleDebugInfo { module: *module }, source)),
-        )
-      })
-      .collect::<Result<Vec<_>>>()?;
+    // Prepare data for parallel processing, instead of collecting them immediately
+    let mut sources_to_process = Vec::new();
+
+    for &module in ordered_css_modules {
+      let module_id = module.identifier();
+      let code_gen_result = compilation
+        .code_generation_results
+        .get(&module_id, Some(&chunk.runtime));
+
+      if let Some(code) = code_gen_result.get(&SourceType::Css) {
+        // Determine the order of the modules by whether it is 'css-import'
+        let priority =
+          if let Some(external_module) = module.as_any().downcast_ref::<ExternalModule>() {
+            // If the module is an external module and its type is 'css-import', give it a higher priority (0)
+            if external_module.get_external_type() == "css-import" {
+              0
+            } else {
+              1
+            }
+          } else {
+            // If the module is not an external module, give it a lower priority (1)
+            1
+          };
+
+        // Push the module into the list for processing, along with its priority and code
+        sources_to_process.push((priority, CssModuleDebugInfo { module }, code.clone()));
+      }
+    }
+
+    // Sort all modules in order according to their priority attribute
+    sources_to_process.sort_by_key(|&(priority, _, _)| priority);
```

Comment:
order again the `ordered_css_modules` seems wired

`ExternalModule::source_types` should be `SourceType::CssImport`, currently it's `SourceType::Css`, after change it to `SourceType::CssImport`, it should be easy to sort it in `CssPlugin::get_ordered_chunk_css_modules`

Reply:
Thank you for the guidance. I will try it out according to your advice.

Reply:
I have revised the code according to your suggestions and added test cases. Could you please check if there are any further issues with the code? Thank you.